C-Sharp.md

```diff
@@ -1,6 +1,268 @@
-#The C# Programming Language
+# The C# Programming Language
 
+1. ### Basics
+  * Setup
+  * Your first C# program
+  * Types and variables
+  * Flow control statements
+  * Operators
+  * Strings
+  * Classes, Objects, Interface and Main methods
+  * Fields and properties
+  * Scope and accessibility modifiers
+  * Handling exceptions
 
-* Introduction
-* Setup
-* Language features 
+2. ### Intermediate
+  * Generics
+  * Events, delegates and lambda expressions
+  * Collection framework
+  * LINQ
+
+3. ### Advanced
+  * Asynchronous Programming (Async and Await)  
+  * Task Parallel Library
+
+5. ### What's New in C# 6
+  * Null-Conditional operator
+  * Auto-Property initializers
+  * Nameof expressions
+  * Expression bodied functions and properties
+  * Other features
+
+4. ### Object-Oriented principles (OOP)
+   * Encapsulation
+   * Abstraction
+   * Inheritance
+   * Polymorphism
+
+6. ### Solid principles
+  * Single responsibility principle
+  * Open closed principle
+  * Liskov substitution principle
+  * Interface segregation principle
+  * Dependency inversion principle
+
+7. ### C# Best practices, Design patterns & Test driven development (TDD)
+
+### Setup
+[LinqPad][linqpad] is an ultimate .NET scratchpad to quickly test your C# code snippets.The standard edition is free and a perfect tool for beginners to quickly execute language statements, expressions and programs.
```

Comment:
I would remove the word 'ultimate' as it is subjective. I would also consider changing 'quickly' to 'easily' as speed in this case is also subjective based upon the user.